src/iop/borders.c

```diff
@@ -215,15 +229,58 @@ int legacy_params(dt_iop_module_t *self,
+    dt_iop_borders_params_v4_t *n =
+      (dt_iop_borders_params_v4_t *)malloc(sizeof(dt_iop_borders_params_v4_t));
 
     memcpy(n, o, sizeof(struct dt_iop_borders_params_v2_t));
     n->max_border_size = FALSE;
 
     *new_params = n;
-    *new_params_size = sizeof(dt_iop_borders_params_v3_t);
-    *new_version = 3;
+    *new_params_size = sizeof(dt_iop_borders_params_v4_t);
+    *new_version = 4;
+    return 0;
+  }
+
+  if(old_version == 3)
+  {
+    typedef struct dt_iop_borders_params_v3_t
+    {
+      float color[3];           // border color $DEFAULT: 1.0
+      float aspect;             /* aspect ratio of the outer frame w/h
+                                $MIN: 1.0 $MAX: 3.0 $DEFAULT: DT_IOP_BORDERS_ASPECT_CONSTANT_VALUE $DESCRIPTION: "aspect ratio" */
+      char aspect_text[20];     /* UNUSED aspect ratio of the outer frame w/h (user string version)
+                                  DEFAULT: "constant border" */
+      dt_iop_orientation_t aspect_orient;        /* aspect ratio orientation
+                                                    $DEFAULT: 0 $DESCRIPTION: "orientation" */
+      float size;               /* border width relative to overall frame width
+                                  $MIN: 0.0 $MAX: 0.5 $DEFAULT: 0.1 $DESCRIPTION: "border size" */
+      float pos_h;              /* picture horizontal position ratio into the final image
+                                  $MIN: 0.0 $MAX: 1.0 $DEFAULT: 0.5 $DESCRIPTION: "horizontal offset" */
+      char pos_h_text[20];      /* UNUSED picture horizontal position ratio into the final image (user string version)
+                                  DEFAULT: "1/2" */
+      float pos_v;              /* picture vertical position ratio into the final image
+                                  $MIN: 0.0 $MAX: 1.0 $DEFAULT: 0.5 $DESCRIPTION: "vertical offset"*/
+      char pos_v_text[20];      /* UNUSED picture vertical position ratio into the final image (user string version)
+                                  DEFAULT: "1/2" */
+      float frame_size;         /* frame line width relative to border width
+                                  $MIN: 0.0 $MAX: 1.0 $DEFAULT: 0.0 $DESCRIPTION: "frame line size" */
+      float frame_offset;       /* frame offset from picture size relative to [border width - frame width]
+                                  $MIN: 0.0 $MAX: 1.0 $DEFAULT: 0.5 $DESCRIPTION: "frame line offset" */
+      float frame_color[3];     // frame line color $DEFAULT: 0.0
+      gboolean max_border_size; /* the way border size is computed
+                                  $DEFAULT: TRUE */
+    } dt_iop_borders_params_v3_t;
+
+    const dt_iop_borders_params_v3_t *o = (dt_iop_borders_params_v3_t *)old_params;
+    dt_iop_borders_params_v4_t *n =
+      (dt_iop_borders_params_v4_t *)malloc(sizeof(dt_iop_borders_params_v4_t));
+
+    memcpy(n, o, sizeof(struct dt_iop_borders_params_v3_t));
+    n->basis = DT_IOP_BORDERS_BASIS_AUTO;
```

Comment:
Can we do the proper migration here? Based on aspect_orient & max_border_size we should be able to set basis to new `DT_IOP_BORDERS_BASIS_WIDTH`, `DT_IOP_BORDERS_BASIS_LONGEST` or `DT_IOP_BORDERS_BASIS_WIDTH`.